### PR TITLE
Modification of K-Means Stopping Criteria

### PR DESCRIPTION
**Pull request by owerbat**
Stopping criteria in K-Means modified to be consistent with stock Scikit-learn. Also one Scikit-learn test has been added to deselected because of instability of the algorithm.

### CODE REVIEW

**Comment by masdevas** [comment]
I think that we should use different "default epsilon" for float and double data types because of different precision. Also, if PR changes our behaviour under sklearn interfaces, @SmirnovEgorRu should check this request.

**Comment by SmirnovEgorRu** [comment]
I suppose we need to change stopping criteria in DAAL itself, not to have a dirty hack on Scikit-learn patching side.

File: deselected_tests.yaml [hunk 1]
```diff
@@ -72,3 +72,7 @@ deselected_tests:
   # We have difference (Max absolute difference: 1.97215226e-31) in computing of log_proba with np.log(y_proba)
   # Looks like using IDP NumPy is the cause for the failure due to use of more aggressive compiler optimization when compile NumPy UFunc loops
   - neural_network/tests/test_mlp.py::test_predict_proba_multilabel
+
+  # Different number of iterations because of instability of kmeans
```
Comment (PetrovKP) [inline deselected_tests.yaml:76]: This is a bug, can we fix it?

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 2]
```diff
@@ -123,11 +123,13 @@ def is_string(s, target_str):
 def _daal4py_kmeans_compatibility(nClusters, maxIterations, fptype = "double",
     method = "lloydDense", accuracyThreshold = 0.0, resultsToEvaluate = "computeCentroids"):
     kmeans_algo = None
+    accuracyThreshold = 1e-6 if accuracyThreshold == 0.0 else accuracyThreshold
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:126]: We used a tolerance equal to 0. This logic should be left.

**Review by masdevas — APPROVED** [review]

**Review by PetrovKP — APPROVED** [review]

**Pull request merged**